packages/domains-table/src/domains-table/domains-table.tsx

```diff
@@ -362,7 +380,26 @@ export const DomainsTable = ( props: DomainsTableProps ) => {
 		handleRestartDomainStatusPolling,
 		showBulkActions,
 		setShowBulkActions,
-		onDomainAction,
+		onDomainAction: async ( actionType, domain ) => {
+			const actionDescription = onDomainAction?.( actionType, domain );
+
+			if ( ! actionDescription ) {
+				return;
+			}
+
+			const { message, action } = actionDescription;
+
+			setUpdatingDomain( { domain: domain.domain, message } );
+
+			try {
+				await action();
+			} catch {
+				// Handle this error
```

Comment:
I don't think this `try...catch` is necessary. There is already error handling in the actions. I tried a couple of error scenarios - disable network connection, and modify API to return error - and we show a notice already for both of them.